internal/operands/node-labeller/reconcile.go

```diff
@@ -0,0 +1,126 @@
+	}
+}
+
+func WatchClusterTypes() []runtime.Object {
+	return []runtime.Object{
+		&rbac.ClusterRole{},
+		&rbac.ClusterRoleBinding{},
+		&secv1.SecurityContextConstraints{},
+	}
+}
+
+func Reconcile(request *common.Request) error {
+	for _, f := range []func(*common.Request) error{
+		reconcileClusterRole,
+		reconcileServiceAccount,
+		reconcileClusterRoleBinding,
+		reconcileConfigMap,
+		reconcileDaemonSet,
+		ReconcileSecurityContextConstraint,
+	} {
+		if err := f(request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Cleanup(request *common.Request) error {
+	for _, obj := range []controllerutil.Object{
+		newClusterRole(request.Namespace),
+		newClusterRoleBinding(request.Namespace),
+	} {
+		err := request.Client.Delete(request.Context, obj)
+		if err != nil && !errors.IsNotFound(err) {
+			request.Logger.Error(err, fmt.Sprintf("Error deleting \"%s\": %s", obj.GetName(), err))
+			return err
+		}
+	}
+	return nil
+}
+
+func reconcileClusterRole(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request,
+		newClusterRole(request.Namespace),
+		func(newRes, foundRes controllerutil.Object) {
+			foundRes.(*rbac.ClusterRole).Rules = newRes.(*rbac.ClusterRole).Rules
+		})
+}
+
+func reconcileServiceAccount(request *common.Request) error {
+	return common.CreateOrUpdateResource(request,
+		newServiceAccount(request.Namespace),
+		func(_, _ controllerutil.Object) {})
+}
+
+func reconcileClusterRoleBinding(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request,
+		newClusterRoleBinding(request.Namespace),
+		func(newRes, foundRes controllerutil.Object) {
+			newBinding := newRes.(*rbac.ClusterRoleBinding)
+			foundBinding := foundRes.(*rbac.ClusterRoleBinding)
+			foundBinding.RoleRef = newBinding.RoleRef
+			foundBinding.Subjects = newBinding.Subjects
+		})
+}
+
+func reconcileConfigMap(request *common.Request) error {
+	return common.CreateOrUpdateResource(request,
+		newConfigMap(request.Namespace),
+		func(newRes, foundRes controllerutil.Object) {
+			newConfigMap := newRes.(*v1.ConfigMap)
+			foundConfigMap := foundRes.(*v1.ConfigMap)
+			foundConfigMap.Data = newConfigMap.Data
+		})
+}
+
+func reconcileDaemonSet(request *common.Request) error {
+	nodeLabellerSpec := &request.Instance.Spec.NodeLabeller
+	daemonSet := newDaemonSet(request.Namespace)
+	addPlacementFields(daemonSet, &nodeLabellerSpec.Placement)
+	return common.CreateOrUpdateResource(request,
+		daemonSet,
+		func(newRes, foundRes controllerutil.Object) {
+			foundRes.(*apps.DaemonSet).Spec = newRes.(*apps.DaemonSet).Spec
+		})
+}
+
+func addPlacementFields(daemonset *apps.DaemonSet, nodePlacement *lifecycleapi.NodePlacement) {
+
+	podSpec := &daemonset.Spec.Template.Spec
+	podSpec.Affinity = nodePlacement.Affinity
+	podSpec.NodeSelector = nodePlacement.NodeSelector
+	podSpec.Tolerations = nodePlacement.Tolerations
+}
+
+func ReconcileSecurityContextConstraint(request *common.Request) error {
```

Comment:
Any reason this function is exported?

Reply:
No, it was an oversight :). Changed it. Thanks!